### ISSUE TITLE
Reload Content using ajax - remove all events

### ISSUE THREAD

**Issue by totpero** [Flyer53/jsPanel4#201]
Hi,

Related to this issue #132 , 
I have all the modal events binded on `panel.content` element.
 After I reload panel content using ajax and  when I try to rebind all events to  `panel.content` , the events are duplicated.
I want to off all events attached to the `panel.content` after ajax reload.
I have try also this on done:

```
const clonedElement = panel.content.cloneNode(true);
clonedElement.innerHTML = xhr.responseText;
panel.content.replaceWith(clonedElement);
```
or
`jsPanel.emptyNode(that._panel.content);`

Can you off all events attached to the  `panel.content` after ajax reload or after content replaced?
Thanks,
The jsPanel4 is awesome project 👍

**Comment by Flyer53** [comment]
The property `panel.content` refers to `<div class="jsPanel-content">` which is the element containing all the content created by you. It's a very basic part of the default panel structure and you should not attach your event handlers to this element because it is not meant to be removed. When you remove panel content the child elements of `<div class="jsPanel-content">` are removed.

Attach your event handlers to the elements you add t ot the panel. Then you don't have th remove them again when you empty the panel with either `panel.contentRemove()` or `jsPanel.emptyNode(that._panel.content)` because the events are removed with the elements they are attached to.

When, for whatever reason, you insist on attaching event handlers to `<div class="jsPanel-content">` using
[EventTarget.addEventListener()](https://developer.mozilla.org/en-US/docs/Web/API/EventTarget/addEventListener)
you have to remove the event handlers manually using
[EventTarget.removeEventListener()](https://developer.mozilla.org/en-US/docs/Web/API/EventTarget/removeEventListener)
each time you empty the panel or remove some content with an event handler attached.

**_In short:_**
Attach your event handlers to the stuff you load into the panel. **Not to the property `panel.content`.** Then it should work fine when you reload content.

I hope that helps.
Regards,
Stefan

**Comment by totpero** [comment]
Hi @Flyer53 ,
Thanks for your answer.
I attache the events on `jsPanel-content` and filter the selector like this:
` $(that._panel.content).on('click', '#element',....`
because the elements is not loaded immediately after the ajax reload call and the elements don't exists in the dom...
Thanks

**Comment by Flyer53** [comment]
@totpero For some reason your last answer doesn't show up on my PC ...

However, you should be able to reload content without adding the event handers once more. Or if you don't want to do that remove the handlers before you reload content. See https://api.jquery.com/off/#off-events-selector-handler since you use jQuery.

Still I don't see a benefit in adding all the handlers to `<div class="jsPanel-content">`. Adding the handlers to the targets itself would avoid all this ...

Ultimately this is not a jsPanel issue, but a matter of how you organize loading/reloading content and adding your event handlers.

**Comment by totpero** [comment]
Thank @Flyer53 for your comment, i will try to add the events directly  to the element (input, button, etc) is the best solution to avoid this problem. 
Thanks and you do good job with this library.
